src/Features/CSharp/Portable/Completion/KeywordRecommenders/AssemblyKeywordRecommender.cs

```diff
@@ -17,15 +15,40 @@ public AssemblyKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            if (context.IsTypeAttributeContext(cancellationToken))
+            var token = context.TargetToken;
+
+            if (token.Kind() != SyntaxKind.OpenBracketToken)
             {
-                var token = context.LeftToken;
-                var type = token.GetAncestor<MemberDeclarationSyntax>();
+                return false;
+            }
 
-                return type == null || type.IsParentKind(SyntaxKind.CompilationUnit);
+            if (token.Parent.Kind() == SyntaxKind.AttributeList)
+            {
+                var attributeList = token.Parent;
+                var parentSyntax = attributeList.Parent;
+                switch (parentSyntax)
+                {
+                    case CompilationUnitSyntax _:
+                    // There are some case the parente of attributeList is (Class/Interface/Enum/Struct)DeclarationSyntax, like:
+                    // [$$
+                    // class Goo {
+                    // for these cases is necessary check if they Parent is CompilationUnitSyntax
+                    case BaseTypeDeclarationSyntax baseType when baseType.Parent is CompilationUnitSyntax:
+                    // There is some case the parente of attributeList is IncompleteMemberSyntax(See test: TestInOuterAttribute), like:
```

Comment:
>There is some case the parente of attributeList is IncompleteMemberSyntax(See test: TestInOuterAttribute), like: [](start = 23, length = 112)

The case where the parent of attributeList is IncompleteMemberSyntax(See test: <see cref="TestInOuterAttribute">), like: